### ISSUE TITLE
How can I use the use the compiled js files instead of the src files?

### ISSUE THREAD

**Issue by jdjuan** [jvandemo/generator-angular2-library#45]
Hey @jvandemo , this is a great and probably the best generator for Angular 2 libraries out there. You have done a great job. So thanks.

Second, I created a package and installed in my Angular 2 web application. Now when I import it, it seems like it is loading the .ts files instead of the .js generated bundle. I guess we actually want to used the compiled files instead? Or did you meant to export the .ts files to allow the user to create one unique bundle at the end?

PS: I didn't know if I should publish the question here or in StackOverflow, if needed I can delete this issue and migrate the question there.

**Comment by caroso1222** [comment]
@jdjuan, 
**TL;DR** 
It's actually using the transpiled `.js` files because of the `main` property in `package.json`. That's where Webpack is resolving the actual import. 

**For the curious...**
Try this to check it yourself:

1. Create app with generator: `yo angular2-library`. Let's say you go with name `ngx-library`.
2. Eventhough the `dist` folder is generated, go ahead and transpile files again: `npm run tsc`
3. Create a new Angular app. Let's assume you use angular-cli. `ng new hello-world`
4. Run `npm link` inside `/ngx-library`, and `npm link ngx-library` inside `/hello-world` to have your library available inside your Angular app `node_modules` folder.
5. Import your new `SampleModule` module into `app.module` (add to decorator imports as well). If you have problems when running `ng serve`, try adding the library to decorator as `SampleModule.forRoot()`.
6. Create `<sample-component></sample-component>` in `app.component.html`. Run `ng serve`. It should show an h1 with text "Sample Component". Our library is working.
7. Now go back to `/ngx-library` and change `sample.component.ts` to have a different selector, say `selector: 'sample1-component'`. No need to npm link again, it's automatically update. **Don't transpile yet.**
8. Refresh browser. Angular shouldn't be complaining about `sample-component` not being declared, because its references are the compiled files. **This is where you realize you're actually using the transpiled `.js` files because `sample.component.ts` no longer declares 'sample-component' as its selector.**
9. Transpile library again `npm run tsc`.
10. Refresh browser. Angular should now complain about `sample-component` not being declared.
11. Change `<sample-component></sample-component>` to `<sample1-component></sample1-component>`. 
12. App should work now and 'Sample Component' showing up.

**For the more curious...**
Try removing `"main": "./dist/index.js"` from `package.json` and change the same component selector. The `SampleModule` import now resolves to the source `.ts` files.

**Comment by jdjuan** [comment]
Awesomeeeeeee!, you should get a reward for this explanation. Thank you very much @caroso1222 this solves my initial concern.
Hopefully next time opens a related issues it can be referenced here. (y)

**Comment by jvandemo** [comment]
@caroso1222 &mdash; Thank you so much for writing that down. This will help clear up a lot of confusion 👍 

@jdjuan &mdash; The files that are used depend indeed on the environment that you are in. Some environments use other keys from your `package.json` file than others (e.g. node, webpack, TypeScript, Closure, etc).

The Angular team is currently working on figuring this out and as soon as there are official recommendations, we will also update this generator to reflect them.

Thanks again for your feedback, much appreciated!
